engine.go

```diff
@@ -99,7 +99,7 @@ func (w workflowEngine) Parse(
 		return nil, err
 	}
 	if stepWorkflowFileCache != nil {
-		files = loadfile.MergeFileCaches(stepWorkflowFileCache, files)
+		files, _ = loadfile.MergeFileCaches(stepWorkflowFileCache, files)
```

Comment:
Granted, we don't have a current use case where this will fail: but, if we do, we're setting `files` to `nil` and the `executor.Prepare` call below will presumably fail with a dereference error on `files.Contents()`. This doesn't seem like the right thing to do.

I think either we need to look at the error here and fail out or, alternately, the merge could treat the mismatched roots as a "warning" and issue a diagnostic while preserving the original root and returning a file cache with that value so you can "carry on".

Reply:
Thanks.